sentinel-adapter/sentinel-spring-webmvc-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/spring/webmvc/SentinelInterceptor.java

```diff
@@ -0,0 +1,134 @@
+ *
+ * @Author kaizi2009
+ */
+public class SentinelInterceptor extends AbstractSentinelInterceptor {
+    private static final String EMPTY_ORIGIN = "";
+    private final static String COLON = ":";
+    private SentinelSpringMvcConfig config;
+
+    public SentinelInterceptor(SentinelSpringMvcConfig config) {
+        if (config == null) {
+            throw new SentinelSpringMvcException("Config can not be null");
+        }
+        this.config = config;
+    }
+
+    public SentinelInterceptor() {
+        this(new SentinelSpringMvcConfig());
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+            throws Exception {
+        Entry urlEntry = null;
+        Entry httpMethodUrlEntry = null;
+
+        try {
+            String resourceName = getResourceName(request);
+
+            UrlCleaner urlCleaner = config.getUrlCleaner();
+            if (urlCleaner != null) {
+                resourceName = urlCleaner.clean(resourceName);
+            }
+
+            if (!StringUtil.isEmpty(resourceName)) {
+                // Parse the request origin using registered origin parser.
+                String origin = parseOrigin(request);
+                ContextUtil.enter(SPRING_MVC_CONTEXT_NAME, origin);
+                urlEntry = SphU.entry(resourceName, EntryType.IN);
+                // Add method specification if necessary
+                if (config.isHttpMethodSpecify()) {
+                    httpMethodUrlEntry = SphU.entry(request.getMethod().toUpperCase() + COLON + resourceName,
+                            EntryType.IN);
+                }
+            }
+            final EntryContainer entryContainer = new EntryContainer().setUrlEntry(urlEntry)
+                    .setHttpMethodUrlEntry(httpMethodUrlEntry);
+            setEntryContainerInReqeust(request, config.getRequestAttributeName(), entryContainer);
+            return true;
+        } catch (BlockException e) {
+            //Throw BlockException and handle it in spring MVC
+            throw e;
+        } catch (RuntimeException e1) {
+            Tracer.traceEntry(e1, urlEntry);
+            Tracer.traceEntry(e1, httpMethodUrlEntry);
+            throw e1;
+        }
+    }
+
+    @Override
+    public void afterCompletion(HttpServletRequest request, HttpServletResponse response,
+                                Object handler, Exception ex) throws Exception {
+        EntryContainer entryContainer = getEntryContainerInReqeust(request, config.getRequestAttributeName());
+        if (entryContainer != null) {
+            if (entryContainer.getUrlEntry() != null) {
+                entryContainer.getUrlEntry().exit();
+            }
+            if (entryContainer.getUrlEntry() != null) {
+                entryContainer.getUrlEntry().exit();
+            }
+            removeEntryContainerInReqeust(request, config.getRequestAttributeName());
```

Comment:
First exit httpMethodUrlEntry and then exit urlEntry.
And Trace the `Exception ex ` at here may be better.

Reply:
I will modify it

Reply:
Done